rosbag2_performance/rosbag2_performance_benchmarking/test/benchmark_test.py

```diff
@@ -0,0 +1,418 @@
+    """Rosbag2 benchmark results checker."""
+
+    def __init__(self, root_folder, benchmarks, producers):
+        self.root_folder = root_folder
+        self.results = {}
+        for benchmark in benchmarks:
+            for producer in producers:
+                self.results[(benchmark, producer)] = BenchmarkResults(benchmark, producer)
+
+    def find_benchmark_directories(self):
+        """
+        Return all the benchmark directories in the root folder cases.
+
+        If multiple directories are found only the most recent one is returned.
+        """
+        for (benchmark, producer), result in self.results.items():
+            pattern = benchmark + '_' + producer + '*'
+            files = find_directory(pattern, self.root_folder)
+            if files:
+                # add just the latest one
+                sorted_files = sorted(files)
+                result.directory = sorted_files[-1]
+            else:
+                print('No benchmark directories found in ' + self.root_folder)
+
+    def load_results(self):
+        """Process the benchmark results and generates a report."""
+        self.find_benchmark_directories()
+        for (benchmark, producer), result in self.results.items():
+            print('read_results')
+            if not result.directory:
+                print('No results found for ' + benchmark + ' ' + producer)
+            else:
+                print('Generate report for: ' + result.directory)
+                report = Report(result.directory)
+                result.reports = report.generate()
+
+    def check_expectations(self):
+        """Check the generated report against the expected results and stores the result."""
+        self.load_results()
+
+        print(self.results.items())
+        for (benchmark, producer), result in self.results.items():
+            if not result.reports:
+                print('No results found for ' + benchmark + ' ' + producer)
+                result.test_passed = False
+            else:
+                for report in result.reports:
+                    if check_benchmark_result(report, result.expectation):
+                        print('Test passed: ' + result.directory)
+                        result.test_passed = True
+                    else:
+                        print('Test failed: ' + result.directory)
+                        result.test_passed = False
+
+    def set_expectations(self, benchmark, producer, expectation):
+        self.results[(benchmark, producer)].expectation = expectation
+
+
+def generate_test_description():
+    proc_env = os.environ.copy()
+    proc_env['PYTHONUNBUFFERED'] = '1'
+
+    config_path = os.path.join(
+        ament_index_python.get_package_share_directory('rosbag2_performance_benchmarking'),
+        'config'
+    )
+
+    benchmark_path = os.path.join(config_path, 'benchmarks/test')
+    producers_path = os.path.join(config_path, 'producers')
+
+    bag_root_folder = '/tmpfs/rosbag2_performance_test'
```

Comment:
@MichaelOrlov Currently there are several parameters hardcoded in the integration test. Ideally these should be loaded from YAML files.

1. benchmarks => These are the benchmark YAML files to run. It would be better to load the list of benchmarks we want to run from a `benchmark.yaml` file
2. bag_root_folder => This is where we expect to find the results. It should match with path specified in the "benchmarks" YAML files. One option would be to override the `bag_root_folder` parameter so there is a single source of truth.
3. expectations => These are currently hardcoded to 95%. This is something we could include in the `benchmark.yaml` file

I would add a TODO note above each of these parameters  explaining they will be loaded from a YAML file in the future.

Reply:
@carlossvg I agree that those parameters shall go in some config file but not in benchmark yaml which is using for runing performance benchmarking. It shall be some separate config file for `benchmark_test`.
Anyway I've followed up and added TODOs for this.

Reply:
@MichaelOrlov Yes sorry, that's what I meant, use a separate YAML file. Thanks.